include/common/firmware/i2c_comms.hpp

```diff
@@ -0,0 +1,20 @@
+#pragma once
+#include <span>
+
+#include "stm32g4xx_hal_conf.h"
+
+namespace i2c {
+class I2C {
+  public:
+    uint8_t DEVICE_ADDRESS = 0x1;
+    static constexpr auto BUFFER_SIZE = 1;
+    using BufferType = std::array<uint8_t, BUFFER_SIZE>;
```

Comment:
We might want to make a BufferType concept in a follow-up PR? I decided not to do that here.

Reply:
or make it a span or something